bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/WebrevStorage.java

```diff
@@ -322,26 +322,45 @@ private URI createAndArchive(PullRequest pr, Repository localRepository, Path sc
             throw new UncheckedIOException(e);
         }
     }
-
     interface WebrevGenerator {
         WebrevDescription generate(Hash base, Hash head, String identifier, WebrevDescription.Type type);
 
         WebrevDescription generate(Diff diff, String identifier, WebrevDescription.Type type, String description);
     }
 
-    WebrevGenerator generator(PullRequest pr, Repository localRepository, Path scratchPath, HostedRepositoryPool hostedRepositoryPool) throws IOException {
-        var jsonLocalStorage = jsonStorage == null ? null : hostedRepositoryPool.checkout(jsonStorage, storageRef, scratchPath);
-        var htmlLocalStorage = htmlStorage == null ? null : hostedRepositoryPool.checkout(htmlStorage, storageRef, scratchPath);
+    WebrevGenerator generator(PullRequest pr, Repository localRepository, Path scratchPath, HostedRepositoryPool hostedRepositoryPool) {
 
         return new WebrevGenerator() {
+            Repository jsonLocalStorage = null;
+            Repository htmlLocalStorage = null;
+
+            private void initializeLocalStorage() {
+                if (jsonLocalStorage == null && !(jsonStorage == null)) {
+                    try {
+                        jsonLocalStorage = hostedRepositoryPool.checkout(jsonStorage, storageRef, scratchPath);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
```

Comment:
This would throw `UncheckedIOException` before so should still do that. Can also consider just having one try-catch in the method, surrounding all of it, to avoid repeating the catch clause.

Reply:
You are right, thanks for catching it!